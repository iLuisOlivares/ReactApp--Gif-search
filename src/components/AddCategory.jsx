import React, { useState } from 'react';
import PropTypes from 'prop-types';

function AddCategory({ setCategories }) {

    const [inputValue, setInputValue] = useState("");


    const handleOnChange = (e) => {
        setInputValue(e.target.value);
    }
    const handleSubmit = (e) => {
        e.preventDefault();

        console.log('HandleSubmit:', inputValue);
        if (inputValue.trim().length > 2) {
            setCategories(cats => [e.target[0].value, ...cats]);
            setInputValue('');
        }

    }

    return (
        <form className='search' onSubmit={handleSubmit}>
            <input

                type="text"
                value={inputValue}
                onChange={handleOnChange}
            ></input>
        </form>);

}

export default AddCategory;

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}