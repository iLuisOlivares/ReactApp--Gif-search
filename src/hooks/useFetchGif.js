import { useEffect, useState } from "react";
import { getGifs } from "../helpers/getGif";

export const useFetchGif = (category) => {

    const [state, setState] = useState({
        data: [],
        loading: true

    });

    useEffect(() => {
        getGifs(category)
            .then(img => {
                setState({
                    data: img,
                    loading: false
                })
            })
    }, [category])



    return state // {data: [], loading: true}

};
