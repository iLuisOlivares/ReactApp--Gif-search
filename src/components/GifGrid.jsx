import React, { useState, useEffect } from 'react';
import PropTypes from 'prop-types';
import { getGifs } from '../helpers/getGif';
import { useFetchGif } from '../hooks/useFetchGif';
import { GifGridItem } from './GifGridItem';

export const GifGrid = ({ category }) => {

    // const [images, setimages] = useState([]);

    // useEffect(() => {
    //     getGifs(category).then(img => setimages(img));

    // }, [category]);

    const { data: images, loading } = useFetchGif(category);

    return (
        <>
            <h3 className='animate__animated animate__fadeInUp texto'>{category}</h3>

            {/* {loading && <p className='animate__animated animate__flash'>Loading</p>} */}


            <div className='card-grid '>
                {
                    images.map((img) => (
                        <GifGridItem
                            key={img.id}
                            {...img}
                        ></GifGridItem>
                    ))
                }
            </div>
        </>
    )
}


GifGrid.propTypes = {
    category: PropTypes.string.isRequired
}