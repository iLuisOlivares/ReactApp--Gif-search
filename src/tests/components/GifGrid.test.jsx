import { shallow } from 'enzyme'
import { GifGrid } from '../../components/GifGrid';
import { useFetchGif } from '../../hooks/useFetchGif'
jest.mock('../../hooks/useFetchGif.js')

describe('Test de GifGrid', () => {
    test('Comprobar el snapshot', () => {

        useFetchGif.mockReturnValue({
            data: [],
            loading: true
        })

        const wrapper = shallow(<GifGrid category='Naruto'></GifGrid>)

        expect(wrapper).toMatchSnapshot();
    });

    test('Debe mostrar items al usar useFecthGif', () => {

        const gifs = [{
            id: 'ID1',
            url: 'https://localhost/cualquiercosa.jpg',
            title: 'Gif1'
        }]


        useFetchGif.mockReturnValue({
            data: gifs,
            loading: true
        })

        const wrapper = shallow(<GifGrid category='Naruto'></GifGrid>)

        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length);


    });

});
