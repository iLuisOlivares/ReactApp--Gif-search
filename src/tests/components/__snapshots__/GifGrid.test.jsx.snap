// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test de GifGrid Comprobar el snapshot 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeInUp texto"
  >
    Naruto
  </h3>
  <div
    className="card-grid "
  />
</Fragment>
`;
